test/Common.cpp

```diff
@@ -0,0 +1,379 @@
+    EXPECT_EQ(ucdr_alignment(1, 8), 7);
+}
+
+TEST_F(CommonTest, AppendBuffer)
+{
+    uint8_t buffer1[sizeof(buffer)];
+    uint8_t buffer2[sizeof(buffer)];
+
+    ucdr_append_buffer(&us, buffer1, sizeof(buffer1));
+    EXPECT_EQ(us.size, 2 * (sizeof(buffer) - UCDR_BUFFER_INFO_SIZE));
+
+    ucdr_append_buffer(&us, buffer2, sizeof(buffer2));
+    EXPECT_EQ(us.size, 3 * (sizeof(buffer) - UCDR_BUFFER_INFO_SIZE));
+
+    ucdrBufferInfo binfo = us.buffer_info;
+    EXPECT_EQ(binfo.data, buffer);
+    EXPECT_EQ(binfo.size, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+
+    EXPECT_TRUE(ucdr_next_buffer_info(&binfo));
+    EXPECT_EQ(binfo.data, buffer1);
+    EXPECT_EQ(binfo.size, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+    EXPECT_EQ(binfo.origin, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+
+    EXPECT_TRUE(ucdr_next_buffer_info(&binfo));
+    EXPECT_EQ(binfo.data, buffer2);
+    EXPECT_EQ(binfo.size, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+    EXPECT_EQ(binfo.origin, 2 * (sizeof(buffer) - UCDR_BUFFER_INFO_SIZE));
+
+    EXPECT_FALSE(ucdr_next_buffer_info(&binfo));
+
+    EXPECT_TRUE(ucdr_prev_buffer_info(&binfo));
+    EXPECT_EQ(binfo.data, buffer1);
+    EXPECT_EQ(binfo.size, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+
+    EXPECT_TRUE(ucdr_prev_buffer_info(&binfo));
+    EXPECT_EQ(binfo.data, buffer);
+    EXPECT_EQ(binfo.size, sizeof(buffer) - UCDR_BUFFER_INFO_SIZE);
+
+    EXPECT_FALSE(ucdr_prev_buffer_info(&binfo));
+}
+
+TEST_F(CommonTest, ResetStream)
+{
+    ++us.offset;
+    ++us.iterator;
+
+    ucdr_reset_stream(&us);
+    EXPECT_EQ(us.offset, 0);
+    EXPECT_EQ(us.iterator, buffer);
+
+    uint8_t buffer1[sizeof(buffer)];
+    ucdr_append_buffer(&us, buffer1, sizeof(buffer1));
+    ucdr_next_buffer_info(&us.buffer_info);
+    us.iterator = us.buffer_info.data;
+    us.offset = sizeof(buffer1) - UCDR_BUFFER_INFO_SIZE;
+
+    ucdr_reset_stream(&us);
+    EXPECT_EQ(us.offset, 0);
+    EXPECT_EQ(us.iterator, buffer);
+}
+
+TEST_F(CommonTest, Size)
+{
+    EXPECT_EQ(ucdr_size(&us), us.size);
+}
+
+TEST_F(CommonTest, UsedSize)
+{
+    us.origin = 1;
+    us.offset = 2;
+    EXPECT_EQ(ucdr_used_size(&us), us.offset - us.origin);
+}
+
+TEST_F(CommonTest, RemaningSize)
+{
+    us.origin = 1;
+    us.offset = 2;
+    EXPECT_EQ(ucdr_remaining_size(&us), us.size - (us.offset - us.origin));
+}
+
+TEST_F(CommonTest, CopyStream)
+{
+    uint8_t buffer1_obj[127];
+    uint8_t buffer2_obj[256];
+    uint8_t buffer3_obj[128];
+    std::memset(buffer1_obj, 11, sizeof(buffer1_obj));
+    std::memset(buffer2_obj, 11, sizeof(buffer2_obj));
+    std::memset(buffer3_obj, 11, sizeof(buffer3_obj));
+
+    uint8_t buffer1_dst[127] = {0};
+    uint8_t buffer2_dst[256] = {0};
+    uint8_t buffer3_dst[128] = {0};
+
+    ucdrStream us_dst;
+    ucdr_init_stream(&us_dst, buffer1_dst, sizeof(buffer1_dst));
+    ucdr_append_buffer(&us_dst, buffer2_dst, sizeof(buffer2_dst));
+    ucdr_append_buffer(&us_dst, buffer3_dst, sizeof(buffer3_dst));
+
+    uint8_t buffer1_src[128];
```

Comment:
Maybe I didn't understand the test, but it seems like the size should be 127 instead of 128, as the `_obj` and `_dst` related buffers.

Reply:
It checks than streams are properly copied even if they have different buffer forms.